### ISSUE TITLE
Eslint checkstyle output is invalid

### ISSUE THREAD

**Issue by thom-nic** [benmosher/eslint-plugin-import#364]
This module reports `NaN` as a column for some errors which I believe is causing JetBrains' products (Rubymine, Webstorm) eslint integration to choke.  

As it happens, this file is not using ES6 imports, it's using `require` statements.  Regardless, I think this plugin is not reporting a valid error.

Example output:

```
/usr/local/bin/node /Users/thom/dev/voltserver/ops/client/node_modules/eslint/bin/eslint.js --format=checkstyle --stdin --stdin-filename karma.conf.js
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/Users/thom/dev/voltserver/ops/client/karma.conf.js"><error line="1" column="NaN" severity="error" message="Resolve error: Path must be a string. Received null (import/no-unresolved)" source="eslint.rules.import/no-unresolved" /><error line="2" column="22" severity="error" message="Unable to resolve path to module &apos;path&apos;. (import/no-unresolved)" source="eslint.rules.import/no-unresolved" /></file></checkstyle>

Process finished with exit code 1
```

Screenshot for completeness:

<img width="959" alt="screen shot 2016-06-02 at 2 02 33 pm" src="https://cloud.githubusercontent.com/assets/95562/15755648/bdbc8c0e-28cb-11e6-9bec-4eb1c6933ff9.png">

**Comment by benmosher** [comment]
Can you post a gist that reproduces the issue?

I've not seen this, and I use checkstyle for CI.

**Comment by thom-nic** [comment]
Maybe your CI doesn't choke on `NaN` column values :D

Interestingly, to reproduce it all I needed was one line:

```javascript
const child = require('child_process');
```
I think it was actually triggered after I changed the `import/resolver: webpack` in my eslintrc.  Which makes sense it can't find node modules anymore.

Also if I uninstall the `eslint-import-resolver-node` package then run eslint -- or if I give an invalid resolver config, e.g. `import/resolver: blah`, I get a similar error with `NaN` column:

```
± $(npm bin)/eslint --format=checkstyle test.js

<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="/Users/thom/dev/voltserver/ops/client/test.js"><error line="1" column="NaN" severity="error" message="Resolve error: unable to load resolver &quot;blah&quot;. (import/no-unresolved)" source="eslint.rules.import/no-unresolved" /><error line="1" column="7" severity="error" message="&apos;child&apos; is defined but never used (no-unused-vars)" source="eslint.rules.no-unused-vars" /><error line="1" column="23" severity="error" message="Unable to resolve path to module &apos;http&apos;. (import/no-unresolved)" source="eslint.rules.import/no-unresolved" /></file></checkstyle>
```

Compare to using the default formatter, it shows col 0, not `NaN`:
```
± $(npm bin)/eslint test.js

/Users/thom/dev/voltserver/ops/client/test.js
  1:0   error  Resolve error: unable to load resolver "blah"  import/no-unresolved
  1:7   error  'child' is defined but never used              no-unused-vars
  1:23  error  Unable to resolve path to module 'http'        import/no-unresolved
```

Since most of my sources are webpack-based (reactJS app) but a few are not, is it possible to override eslint's `import/resolver` setting for select files?  Regardless, it would seem `NaN` should be coerced to 0 for column value?

**Comment by benmosher** [comment]
Ahhh, interesting. I think I understand now.

Not clear to me if this is truly an ESLint, checkstyle formatter, or plugin issue, but I think I know how to hotfix it in the plugin. Will do.

**Comment by DavidSunny** [comment]
+1

**Comment by betalb** [comment]
@benmosher Was able to reproduce similar issue with webpack-resolver

webpack.config.js contains let/const/etc... but no `'use strict';` statement, because I was running everything on node 6, but when I switched to 4th node, it started to include lint errors of webpack.config file into output with column 'NaN'

Here are examples of errors

```
<error line="1" column="NaN" severity="error"
  message="Resolve error: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode (import/no-absolute-path)"
  source="eslint.rules.import/no-absolute-path"/>
<error line="1" column="NaN" severity="error"
  message="Resolve error: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode (import/no-unresolved)"
  source="eslint.rules.import/no-unresolved"/>
```
